docs/documentation/zh/get-started/TS for JS Programmers.md

```diff
@@ -0,0 +1,289 @@
+  id: 0,
+};
+```
+
+由于 JavaScript 支持类和面向对象编程，TypeScript 也支持。你可以将接口声明与类一起使用：
+
+```ts twoslash
+interface User {
+  name: string;
+  id: number;
+}
+
+class UserAccount {
+  name: string;
+  id: number;
+
+  constructor(name: string, id: number) {
+    this.name = name;
+    this.id = id;
+  }
+}
+
+const user: User = new UserAccount("Murphy", 1);
+```
+
+您可以使用接口对参数进行注释，并将值返回给函数：
+
+```ts twoslash
+// @noErrors
+interface User {
+  name: string;
+  id: number;
+}
+// ---分割线---
+function getAdminUser(): User {
+  //...
+}
+
+function deleteUser(user: User) {
+  // ...
+}
+```
+
+JavaScript 中已经有一些基本类型可用：`boolean`、 `bigint`、 `null`、`number`、 `string`、 `symbol ` 和 `undefined`，它们都可以在接口中使用。TypeScript 将此列表扩展为更多的内容，例如 `any` （允许任何类型）、[`unknown`](/play#example/unknown-and-never) （确保使用此类型的人声明类型是什么）、 [`never`](/play#example/unknown-and-never) （这种类型不可能发生）和 `void` （返回 `undefined` 或没有返回值的函数）。
+
+构建类型有两种语法： [接口和类型](/play/?e=83#example/types-vs-interfaces)。 你应该更喜欢 `interface`。当需要特定功能时使用 `type` 。
+
+## 组合类型
+
+使用 TypeScript，可以通过组合简单类型来创建复杂类型。有两种流行的方法可以做到这一点：联合和泛型。
+
+### 联合
+
+使用联合，可以声明类型可以是许多类型中的一种。例如，可以将 `boolean` 类型描述为 `true` 或 `false` ：
+
+```ts twoslash
+type MyBool = true | false;
+```
+
+_注意：_如果将鼠标悬停在上面的 `MyBool` 上，您将看到它被归类为 `boolean`。这是结构化类型系统的一个属性。下面有更加详细的信息。
+
+联合类型的一个流行用法是描述 `string` 或者 `number` 的[字面量](/docs/handbook/2/everyday-types.html#literal-types)的合法值。
+
+```ts twoslash
+type WindowStates = "open" | "closed" | "minimized";
+type LockStates = "locked" | "unlocked";
+type PositiveOddNumbersUnderTen = 1 | 3 | 5 | 7 | 9;
+```
+
+联合也提供了一种处理不同类型的方法。例如，可能有一个函数处理 `array` 或者 `string`：
+
+```ts twoslash
+function getLength(obj: string | string[]) {
+  return obj.length;
+}
+```
+
+要了解变量的类型， 使用 `typeof`：
+
+| 类型      | 推断语句                           |
+| --------- | ---------------------------------- |
+| string    | `typeof s === "string"`            |
+| number    | `typeof n === "number"`            |
+| boolean   | `typeof b === "boolean"`           |
+| undefined | `typeof undefined === "undefined"` |
+| function  | `typeof f === "function"`          |
+| array     | `Array.isArray(a)`                 |
+
+例如，你可以使函数根据传递的是字符串还是数组返回不同的值：
+
+<!-- prettier-ignore -->
+```ts twoslash
+function wrapInArray(obj: string | string[]) {
+  if (typeof obj === "string") {
+    return [obj];
+//          ^?
+  }
+  return obj;
+}
+```
+
+### 泛型
+
+泛型为类型提供变量。一个常见的例子是数组。没有泛型的数组可以包含任何内容。带有泛型的数组可以描述数组包含的值。
+
+```ts
+type StringArray = Array<string>;
+type NumberArray = Array<number>;
+type ObjectWithNameArray = Array<{ name: string }>;
+```
+
+你可以声明自己使用泛型的类型：
+
+```ts twoslash
+// @errors: 2345
+interface Backpack<Type> {
+  add: (obj: Type) => void;
+  get: () => Type;
+}
+
+// 这一行是一个简写，可以告诉 TypeScript 有一个常量，叫做`backpack`，并且不用担心它是从哪
+// 里来的。
+declare const backpack: Backpack<string>;
+
+// 对象是一个字符串，因为我们在上面声明了它作为 Backpack 的变量部分。
+const object = backpack.get();
+
+// 因为 backpack 变量是一个字符串，不能将数字传递给 add 函数。
+backpack.add(23);
+```
+
+## 结构化的类型系统（structural type system）
```

Comment:
结构化类型系统？

Reply:
@Kingwl 
同上……这就是我加上英文原文的原因。
可以在[另一篇文章](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-oop.html)里看到类似表述：
```
...The types we wrote in the code are present at runtime, and the types are related via their declarations, not their structures.
```
```markdown
It does so by examining the _structure_ of `{ k: 10 }` and `class Empty { }`.
We can see that `{ k: 10 }` has _all_ of the properties that `Empty` does, because `Empty` has no properties.
Therefore, this is a valid call!
```
```markdown
Again, this isn't an error because the _structures_ of these classes are the same.
While this may seem like a potential source of confusion, in practice, identical classes that shouldn't be related are not common.
```
所以这里 `structural` 就是 `structure` 的形容词，没有歧义。
作者在这篇文章里面所作的探讨是十分深入的，虽然我并不是特别理解为什么要使用“结构”一词描述这种关系，但是我确实没有一些好的建议去改动，有时间我试试和作者交流吧，真的学到了挺多hhhh